File: app/src/main/java/com/example/foodorderingapp/Fragment/ProfileFragment.java
package com.example.foodorderingapp.Fragment;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import androidx.fragment.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.foodorderingapp.activity.LoginActivity;
import com.example.foodorderingapp.databinding.FragmentProfileBinding;

public class ProfileFragment extends Fragment {
    private FragmentProfileBinding binding;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        binding = FragmentProfileBinding.inflate(inflater, container, false);

        binding.logOut.setOnClickListener(v -> {
            logoutUser();
            Intent intent = new Intent(requireContext(), LoginActivity.class);
            startActivity(intent);
        });

        return binding.getRoot();
    }

    private void logoutUser() {
        Context context = requireContext();
        context.getSharedPreferences("login", Context.MODE_PRIVATE).edit()
                .clear()
                .apply();
    }
}

File: app/src/main/java/com/example/foodorderingapp/Fragment/ProfileFragment.kt
package com.example.foodorderingapp.Fragment

import android.content.Context
import android.content.Intent
import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.example.foodorderingapp.LoginActivity
import com.example.foodorderingapp.databinding.FragmentProfileBinding

class ProfileFragment : Fragment() {
    private lateinit var binding: FragmentProfileBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        binding = FragmentProfileBinding.inflate(inflater, container, false)
        binding.logOut.setOnClickListener {
            logoutUser()
            val intent = Intent(requireContext(), LoginActivity::class.java)
            startActivity(intent)
        }
        return binding.root
    }

    private fun logoutUser() {
        val sharedPreferences = requireContext().getSharedPreferences("login", Context.MODE_PRIVATE)
        val editor = sharedPreferences.edit()
        editor.clear()
        editor.apply()
    }
}